server/leafnode.go

```diff
@@ -1271,6 +1269,14 @@ func (c *client) processLeafSub(argo []byte) (err error) {
 		return nil
 	}
 
+	acc := c.acc
+	// Check if we have a loop.
+	if string(sub.subject) == acc.getLDSubject() {
```

Comment:
if you check sub.subject to have the prefix leafNodeLoopDetectionSubjectPrefix or even just $, getting the account lock could be avoided for most subjects.

Reply:
Will see if I can optimize a bit here.

Reply:
ok just take a quick look at prefix check.